src/pk_encryption.rs

```diff
@@ -86,6 +89,56 @@ impl PkDecryption {
     pub fn to_bytes(&self) -> Box<[u8; 32]> {
         self.key.to_bytes()
     }
+
+    /// Create a [`PkDecryption`] object by unpickling a PkDecryption pickle in libolm
+    /// legacy pickle format.
+    ///
+    /// Such pickles are encrypted and need to first be decrypted using
+    /// `pickle_key`.
+    pub fn from_libolm_pickle(
+        pickle: &str,
+        pickle_key: &[u8],
+    ) -> Result<Self, crate::LibolmPickleError> {
+        use crate::utilities::unpickle_libolm;
+
+        unpickle_libolm::<PkDecryptionPickle, _>(pickle, pickle_key, PICKLE_VERSION)
+    }
+
+    /// Pickle a [`PkDecryption`] into a libolm pickle format.
+    ///
+    /// This pickle can be restored using the `[PkDecryption::from_libolm_pickle]`
+    /// method, or can be used in the [`libolm`] C library.
+    ///
+    /// The pickle will be encrypted using the pickle key.
+    ///
+    /// ⚠️  ***Security Warning***: The pickle key will get expanded into both
+    /// an AES key and an IV in a deterministic manner. If the same pickle
+    /// key is reused, this will lead to IV reuse. To prevent this, users
+    /// have to ensure that they always use a globally (probabilistically)
+    /// unique pickle key.
+    ///
+    /// [`libolm`]: https://gitlab.matrix.org/matrix-org/olm/
+    ///
+    /// # Examples
+    /// ```
+    /// use vodozemac::pk_encryption::PkDecryption;
+    /// use olm_rs::{pk::OlmPkDecryption, PicklingMode};
+    ///
+    /// let decrypt = PkDecryption::new();
+    ///
+    /// let pickle = decrypt
+    ///     .to_libolm_pickle(&[0u8; 32])
+    ///     .expect("We should be able to pickle a freshly created PkDecryption");
+    ///
+    /// let unpickled = OlmPkDecryption::unpickle(
+    ///     pickle,
+    ///     PicklingMode::Encrypted { key: [0u8; 32].to_vec() },
+    /// ).expect("We should be able to unpickle our exported PkDecryption");
+    /// ```
+    pub fn to_libolm_pickle(&self, pickle_key: &[u8]) -> Result<String, crate::LibolmPickleError> {
```

Comment:
This is currently how Matrix Content Scanner saves it's keys. I have no objections to saving them in a different format going forward. I kept this in out of convenience to minimize changes in the content scanner.

Reply:
I'll go ahead and remove the `to_libolm_pickle` function here :+1:

Reply:
Thanks, that would be great.

Reply:
Hmm I'm not sure if removing this entirely will work. The Matrix Content Scanner backend relies on saving it's key locally in pickled form.

I could change this to be `pickle` and create a pickled PkDecryption which only contains the secret and omits the public key. Either way we will need something that allows us to create a new key and save it somewhere.

@poljar Is there a better way to go about this?

Reply:
Can't you use the `to_bytes()` and `from_bytes()` methods, additionally we could add some `to_base64()` and `from_base64()` methods, or expose a getter to the `Cruve25519SecretKey` which also has such methods.

Reply:
Those would allow us to save the key yes. We would lose the additional encryption we get with the current approach though. (Since the libolm pickle is encrypted with an additional key)

I'm not sure if the encrypted pickle adds much, since the key for it is stored in the Content Scanner's config file.

Thoughts?

Reply:
Oh, you don't have encryption for storage. Let me sleep on that since it's quite late where I'm at.

Reply:
Fuck it, let's keep it then. Somebody else might depend on this as well and then I'll have this conversation again if we don't keep it.

Reply:
For the scanner itself it seems fine to me to just ignore the pickled key and create a new one, it would just break on update for a very small number of reqs that fetched the public key before the update, and it will just work on retry I believe (new public key will be fetch and used).

Reply:
> I'm not sure if the encrypted pickle adds much, since the key for it is stored in the Content Scanner's config file.

Agreed, it seems fairly useless to me.

But poljar point applies, now that it's done here we may as well merge it if others need it, it's not 

Do we need to plug the pickle logic in the content scanner or just remove it altogether (and generate a new key on first start) is another story :)